feat: force pods with volumes to be scheduled on Cloud servers (#743)
hetznercloud/csi-driver: docs/kubernetes/README.md

## Integration with Root Servers

Root servers can be part of the cluster, but the CSI plugin doesn't work there and the current behaviour of the scheduler can cause Pods to be stuck in `Pending`. 

Include topology key evaluation into storage class, by setting the helm chart value `includesRobotServers`:

```yaml
storageClasses:
  - name: hcloud-volumes
    defaultStorageClass: true
    reclaimPolicy: Delete
    includesRobotServers: true # <---
```

Label the nodes according to their type:

**Cloud Servers**
```bash

> kubectl label nodes <node name> instance.hetzner.cloud/is-root-server=false

Reviewer: Considering that we have users with clusters with even more mixed servers (Cloud, Robot, Raspberry Pis at home, ...) I think this nomenclature is confusing.

What do you think about adding a new annotation `instance.hetzner.cloud/{source,type,service,owner}: {cloud,robot}` that we can set in HCCM and use here?

## Integration with Root Servers

Root servers can be part of the cluster, but the CSI plugin doesn't work there and the current behaviour of the scheduler can cause Pods to be stuck in `Pending`. 

> Setting `allowedTopologyCloudServer` to true prevents pods from being scheduled on nodes, specifically Robot servers, where Hetzner volumes are unavailable. This value can not be changed after the initial creation of a storage class.

Reviewer: I would highlight that this option is from the Helm Chart.

```suggestion
In the Helm Chart you can set `allowedTopologyCloudServer` to true to prevent pods from being scheduled on nodes, specifically Robot servers, where Hetzner volumes are unavailable. This value can not be changed after the initial creation of a storage class.
```

```yaml
storageClasses:
  - name: hcloud-volumes
    defaultStorageClass: true
    reclaimPolicy: Delete
    includesRobotServers: true # <---
```

> Label the nodes according to their type:

Reviewer: Should we also add a hint here a la "If you use hcloud-cloud-controller-manager v1.20.0+ these labels are added automatically"?

> To ensure proper topology evaluation, labels are needed to indicate whether a node is a cloud VM or a dedicated server from Robot. If you are using the `hcloud-cloud-controller-manager` version 1.20.0 or later, these labels are added automatically. Otherwise, you will need to label the nodes manually.

Reviewer: Next HCCM version is 1.21.0: https://github.com/hetznercloud/hcloud-cloud-controller-manager/pull/723

```suggestion
To ensure proper topology evaluation, labels are needed to indicate whether a node is a cloud VM or a dedicated server from Robot. If you are using the `hcloud-cloud-controller-manager` version 1.21.0 or later, these labels are added automatically. Otherwise, you will need to label the nodes manually.
```

> ### New Label

Reviewer: ```suggestion
### Adding labels manually
```
